ComponentKit/DataSources/CKCollectionViewTransactionalDataSource.mm

```diff
@@ -93,14 +95,65 @@ - (void)transactionalComponentDataSource:(CKTransactionalComponentDataSource *)d
                   didModifyPreviousState:(CKTransactionalComponentDataSourceState *)previousState
                        byApplyingChanges:(CKTransactionalComponentDataSourceAppliedChanges *)changes
 {
-  [_collectionView performBatchUpdates:^{
-    applyChangesToCollectionView(_collectionView, _attachController, [_componentDataSource state], changes);
-    // Detach all the component layouts for items being deleted
-    [self _detachComponentLayoutForRemovedItemsAtIndexPaths:[changes removedIndexPaths]
-                                                    inState:previousState];
-    // Update current state
-    _currentState = [_componentDataSource state];
-  } completion:NULL];
+  const BOOL changesIncludeNonUpdates = (changes.removedIndexPaths.count ||
+                                         changes.insertedIndexPaths.count ||
+                                         changes.movedIndexPaths.count ||
+                                         changes.insertedSections.count ||
+                                         changes.removedSections.count);
+  const BOOL changesIncludeOnlyUpdates = (!!changes.updatedIndexPaths.count && !changesIncludeNonUpdates);
```

Comment:
`!!changes.updatedIndexPaths.count`? Is that correct? If you intend to check that updated index paths do exist `changes.updatedIndexPaths.count > 0` would be preferable.